src/plugins/dashboard/public/dashboard_app/hooks/use_dashboard_outcome_validation.tsx

```diff
@@ -39,12 +39,15 @@ export const useDashboardOutcomeValidation = () => {
          */
         if (loadOutcome === 'aliasMatch' && dashboardId && alias) {
           const path = scopedHistory.location.hash.replace(dashboardId, alias);
-          if (screenshotMode.isScreenshotMode()) {
-            scopedHistory.replace(path);
-          } else {
-            spaces.redirectLegacyUrl?.({ path, aliasPurpose });
-            return false; // redirected. Stop loading dashboard.
-          }
+          // navigate to alias on the next tick to allow Dashboard to finish rendering its error state.
```

Comment:
Instead of event stack hacking, how about changing the return type of validateOutcome to be a enum instead of a boolean, so you can better define outcomes. For example, really there are three outcomes - found, notFound, and redirect. That way, initializeDashboard and just return early in redirect case instead of going into an error state - which it should not enter.

Reply:
Good idea - let me give that a try.

Reply:
Great suggestion @nreese! I was able to get that working well, just had to propagate the fact that the embeddable create could return `undefined` up the chain. This also let me do at least a jest test verifies that in the event of a `redirect` result, dashboard create should return undefined.